src/lua/httpc.lua

```diff
@@ -228,6 +241,56 @@ local function process_headers(headers)
     return headers
 end
 
+-- In the Extended BNF notation of RFC 1341, a Content-Type header field value
+-- is defined as follows: Content-Type := type "/" subtype *[";" parameter].
+-- See RFC 1341, https://www.w3.org/Protocols/rfc1341/4_Content-Type.html
+-- Function splits content-type header for part with type and subtype and part
+-- parameter and returns the first one.
+local function header_value_wo_parameters(header_value)
+    if header_value == nil then
+        return header_value
+    end
+    if type(header_value) ~= "string" then
+        error('header value must be a string', 2)
+    end
+    local res = string.split(header_value, ';')
+    local header_value = res[1]
+    return header_value:gsub("^%s+", ""):gsub("%s+$", "")
+end
+
+local function encode_body(body, content_type)
+    local default_content_type = 'application/json'
+    local raw_body
+    local body_type = type(body)
+    if body == nil then
+        raw_body = ''
+    elseif body_type == 'cdata' or
+           body_type == 'userdata' or
+           body_type == 'table' then
+        if content_type == nil then
+            content_type = default_content_type
+        end
+        if content_type == nil then
+            error('Unable to encode body: content-type was not specified', 2)
+        end
+        content_type = header_value_wo_parameters(content_type)
+        local encoder = encoders[content_type]
+        if encoder == nil then
+            local msg = 'Unable to encode body: encode function is not found (%s)'
+            error(msg:format(content_type), 2)
+        end
+        local ok, res = pcall(encoder, body, content_type)
```

Comment:
Maybe I am missing something, but why is `content_type` stripped from all parameters before each `encoder` call? Even if current encoders don't know how to handle params, user defined ones might.

Reply:
In your example from doc request:
`Content-Type: text/html; charset=UTF-8`
`charset=URF-8` - is not a part of content type, but it may be crucial for some future encoders.

Reply:
In the recent version, the problem is still the same: we strip `content_type` to `mime_type` and pass `mime_type` to `encoder()`.